js/coinjar.js

```diff
@@ -0,0 +1,411 @@
+            'price': price,
+            'amount': amount,
+            'filled': filled,
+            'remaining': remaining,
+            'trades': undefined,
+            'fee': undefined,
+            'info': order,
+            'cost': undefined,
+            'average': undefined,
+        };
+    }
+
+    async fetchMyTrades (symbol = undefined, since = 0, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'cursor': since,
+        };
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['product_id'] = market['id'];
+        }
+        const response = await this.privateGetFills (this.extend (request, params));
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        const timestamp = this.parse8601 (this.safeString (trade, 'timestamp'));
+        const productId = this.safeString (trade, 'product_id');
+        const symbol = this.markets_by_id[productId]['symbol'];
+        return {
+            'info': trade,
+            'id': this.safeString (trade, 'tid'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': undefined,
+            'side': this.safeString (trade, 'side'),
+            'order': this.safeString (trade, 'oid'),
+            'takerOrMaker': this.safeString (trade, 'liquidity'),
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'size'),
+            'cost': this.safeFloat (trade, 'value'),
+            'fee': this.safeFloat (trade, 'estimated_fee'),
+        };
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'productId': market['id'],
+            'interval': timeframe,
+            'after': since,
+            'before': limit,
+        };
+        const response = await this.publicGetProductsProductIdCandles (this.extend (request, params));
+        // Sample response from publicGetProductsProductIdCandles()
+        // [
+        //     [
+        //         "2018-03-06T00:00:00Z",
+        //         "15010.00000000",
+        //         "15010.00000000",
+        //         "14540.00000000",
+        //         "14550.00000000",
+        //         "8.95300000"
+        //     ],
+        //     [
+        //         "2018-03-06T04:00:00Z",
+        //         "14590.00000000",
+        //         "14700.00000000",
+        //         "14000.00000000",
+        //         "14590.00000000",
+        //         "10.57200000"
+        //     ],
+        // ]
+        return response;
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
```

Comment:
```suggestion
    sign (path, api = 'public', method = 'GET', params = {}, headers = {}, body = undefined) {
```

to resolve thrown @381:
`[TypeError] Cannot set property 'Authorization' of undefined`

Reply:
Thank you. I'll fix it and run the private tests.